packages/babel-plugin-redux-saga/src/index.js

```diff
@@ -1,31 +1,36 @@
 var SourceMapConsumer = require('source-map').SourceMapConsumer
 var pathFS = require('path')
-var importHelper = require('@babel/helper-module-imports')
 
-var fnId = 'reduxSagaSource'
-var tempVarId = 'res'
-var locationSymbolPath = 'redux-saga'
-var locationSymbolExportName = 'SAGA_LOCATION'
-var hintedLocationSymbolExportName = locationSymbolExportName
+var globalSymbolNames = {location: '@@redux-saga/LOCATION'}
+// var globalSymbolNames = require('redux-saga').globalSymbolNames
+
+var wrapperFunctionName = 'reduxSagaSource'
+var tempVarName = 'res'
+var symbolName = globalSymbolNames.location
+var plainPropertyName = globalSymbolNames.location
 
 module.exports = function(babel) {
   var { types: t } = babel
   var sourceMap = null
-  var symbolNameBinding = null
 
-  function getAssignementWithSymbolProperty(name, symbolName) {
-    return t.memberExpression(t.identifier(name), t.identifier(symbolName), true)
+  function getAssignementWithSymbolProperty(name) {
+    return t.memberExpression(
+      t.identifier(name),
+      t.callExpression(
+        t.memberExpression(t.identifier('Symbol'), t.identifier('for')),
+        [t.stringLiteral(symbolName)]
+      ),
+      true);
   }
 
-  function getObjectExtensionNode(path, functionName) {
-    if (!symbolNameBinding) {
-      const { name: locationSymbolName } = importHelper.addNamed(path, locationSymbolExportName, locationSymbolPath, {
-        nameHint: hintedLocationSymbolExportName,
-      })
+  function getAssignementWithPlainProperty(name) { 
```

Comment:
symbols are not enumerable, so maybe we should wrap this fallback output with `Object.defineProperty`?